test/test_unary_ufuncs.py

```diff
@@ -33,6 +202,190 @@ def test_unsupported_dtypes(self, device, dtype, op):
         with self.assertRaises(RuntimeError):
             op(t)
 
+    # Tests bool tensor negation raises the correct error
+    def test_neg_error_message(self, device):
+        self.assertRaisesRegex(
+            RuntimeError,
+            r"Negation, the `\-` operator, on a bool tensor is not supported. "
+            r"If you are trying to invert a mask, use the `\~` or `logical_not\(\)` operator instead.",
+            lambda: - torch.tensor([False, True], device=device))
+
+    @dtypes(*floating_types_and(torch.bfloat16, torch.half))
+    @ops((_fn for _fn in unary_ufuncs if _fn.domain != (None, None)))
```

Comment:
Makes me wonder if there shouldn't be some sort of predicate-y system here...

Reply:
Maybe we should review after a second test suite is added (like binary ufuncs or *like functions) and update it?